frontend/src/state/stateBrowser.js

```diff
@@ -108,9 +121,25 @@ export function dummy() {
   };
 }
 
+export function changeMode(mode, id) {
+  return {
+    type: CHANGE_MODE,
+    mode: mode,
+    id: id,
+  };
+}
+
 /* -- REDUX REDUCER -- */
 const browserReducer = (state = initialState, action) => {
   switch (action.type) {
+    case CHANGE_MODE:
+      return {
+        ...state,
+        inspector: {
```

Comment:
Do ...inspector inside curly brackets to future proof this in case new fields would be added.